IdentifiabilityODE.mpl

```diff
@@ -557,3 +589,139 @@ CompareDiffVar := proc(dvl, dvr, var_list)
+  state_ := parse(cat(StringTools[Join](StringTools[Split](convert(state, string), "_")[..-2], "_"), "_")):
+  if state_ in x_vars then
+    return state_;
+  end if:
+end proc:
+
+#===============================================================================
+GetMinLevelBFS := proc(s, m, x_vars, y_vars, mu, x_eqs, y_eqs_in, all_vars)
+#===============================================================================
+  # this part is copied from original SIAN code
+  local current_level, visible_states, visibility_table, i, j, continue, poly_d,
+  leader, separant, candidates, each, differentiate_, k, y_eqs:
+  y_eqs := y_eqs_in:
+  current_level := 0:
+  # get functions on level 0, we consider parameters and states indistinguishable
+  # i.e. parameters are states with d/dt = 0
+  visible_states :=  foldl(`union`, op(map(x->indets(rhs(x)) minus {t}, y_eqs))); #select(f -> f in x_zero_vars, ); # map(x->parse(convert(x, string)[..-2]), select(f -> f in x_zero_vars, foldl(`union`, op(map(x->indets(rhs(x)), y_eqs))))):# cat(StringTools[Split](convert(x, string), "_")[1], "_")
+
+  # construct a hash table of "visibility"
+  visibility_table := table([seq(GetStateName(each, x_vars, mu)=current_level, each in visible_states)]):
+  # this is a flag array: if i-th position == 1 then we must differentiat i-th y(t) function 
+  differentiate_ := [seq(1, i=1..nops(y_eqs))]: 
+
+  for j from 1 to s + 1 do
+    # begin differentiation
+    current_level := current_level + 1:
+    continue:=true:
+
+    for i from 1 to m do
+      if differentiate_[i]=1 then 
+        poly_d := numer(lhs(y_eqs[i]) - rhs(y_eqs[i])):
+        poly_d := Differentiate(poly_d, all_vars):
+        leader := MakeDerivative(y_vars[i], j):
+        separant := diff(poly_d, leader):
+        poly_d := simplify(leader - subs(x_eqs, -(poly_d - separant * leader) / separant)):
+        y_eqs[i]:= leader = simplify(poly_d - leader);
+        candidates := select(x-> not (GetOrderVar(x)[1]  in y_vars), indets(y_eqs[i])):
+        if op(map(x->not assigned(visibility_table[GetStateName(x, x_vars, mu)]), candidates)) <> NULL then
+          continue := foldl(`or`, op(map(x->not assigned(visibility_table[GetStateName(x, x_vars, mu)]), candidates))):
+        else
+          continue := false;
+        fi:
+        if continue then
+          differentiate_[i]:=1:
+        else
+          differentiate_[i]:=0:
+        fi:
+        for each in candidates do
+          if not assigned(visibility_table[GetStateName(each, x_vars, mu)]) then 
+              visibility_table[GetStateName(each, x_vars, mu)] := current_level:
+          fi:
+        od;
+      fi:
+    od:
+    if add(k, k in differentiate_)=0 then
+        break:
+    fi:
+  od:
+  return visibility_table;
+end proc:
+
+#===============================================================================
+SubsByDepth := proc(sigma, poly_system, poly_vars, non_id, s, m, x_vars, y_vars, mu, x_eqs, y_eqs, all_vars, {trdegsub:=false})

```

Comment:
This function has too many argument. Could you check whether you really need all of them? If yes, then maybe we should pack some of them into a structure/table